src/Grpc.AspNetCore.Server/Internal/HttpContextStreamWriter.cs

```diff
@@ -39,6 +39,14 @@ public WriteOptions WriteOptions
             set => _context.WriteOptions = value;
         }
 
-        public Task WriteAsync(TResponse message) => _context.HttpContext.Response.BodyPipe.WriteMessageAsync(message, _context, _serializer);
+        public Task WriteAsync(TResponse message)
+        {
+            if (message == null)
```

Comment:
Why null check a internal type? That just bloats the method. Just assert.

Reply:
The instance is internal, the interface is public. This method will be called by user code in service methods.